packages/py-api/py_api/controllers/hackathon/participants_controller.py

```diff
@@ -233,11 +233,67 @@ def add_participant(cls, participant: NewParticipant, jwt_token: str | None = No
             return {"message": "New admin participant was registered"}
 
         else:
-            # TODO: create participant
-            # TODO: assign to RANDOM team if there's one with free space or create new one
-            return
+            # Fetch the teams of type random
+            random_teams = TeamFunctionality.fetch_teams_by_condition(
+                {"team_type": "random"},
+            )
 
-        return JSONResponse(
-            content={"message": "The participant was successfully added!"},
-            status_code=200,
-        )
+            # Find the teams which can accept a new participant
+            for team in random_teams:
+
+                if (len(team.team_members) < 6):
+                    avaliable_team = team
+                    try:
+                        # Creates new participant
+                        new_participant = ParticipantsFunctionality.create_participant(
+                            participant,
+                        )
+                        new_participant_object_id = str(
+                            new_participant.inserted_id,
+                        )
+                        # Adds the participant to the team
+                        avaliable_team = TeamFunctionality.add_participant_to_team_object(
+                            avaliable_team.team_name, new_participant_object_id,
+                        )
+                        cls.update_participant(
+                            new_participant_object_id, UpdateParticipant(
+                                **{"team_name": avaliable_team.team_name}
+                            ),
+                        )
+                        TeamFunctionality.update_team_query_using_dump(
+                            avaliable_team.model_dump(),
+                        )
+                        return JSONResponse(content=avaliable_team.model_dump(), status_code=200)
+
+                    except (Exception) as e:
+                        return JSONResponse(content={"message": e.args}, status_code=422)
+
```

Comment:
Add a comment that this defines the flow for creating a random team if all other teams of type random are full